src/test/java/io/vertx/test/core/LogContainsRule.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.test.core;
+
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * A generic log scanner used by testcases to check that messages have appeared
+ * in the JUnit test logs.
+ *
+ */
+public class LogContainsRule implements TestRule {
```

Comment:
we have a similar rule already existing : `io.vertx.test.core.BlockedThreadWarning`, can you reuse this class and perhaps make it more generic to it works in this case too ?

Reply:
Yes, will do.

Reply:
Done. (Hope @tsegismont is OK with the author tag being left as I learned how to do the superclass from this one.)